Spigot-Server-Patches/0384-Anti-Xray.patch

```diff
@@ -208,7 +210,8 @@ index 0000000000000000000000000000000000000000..9c017eb027ce52ea6a2414af2d0190b3
 +
 +            if (blockData != null) {
 +                solidGlobal[i] = blockData.getBlock().isOccluding(blockData, emptyChunk, zeroPos)
-+                    && blockData.getBlock() != Blocks.SPAWNER && blockData.getBlock() != Blocks.BARRIER && blockData.getBlock() != Blocks.SHULKER_BOX;
++                    && blockData.getBlock() != Blocks.SPAWNER && blockData.getBlock() != Blocks.BARRIER && blockData.getBlock() != Blocks.SHULKER_BOX || paperWorldConfig.lavaObscures && blockData == Blocks.LAVA.getBlockData();
++                // Comparing blockData == Blocks.LAVA.getBlockData() instead of blockData.getBlock() == Blocks.LAVA ensures that only "stationary lava" is used
```

Comment:
Comparing against stationary lava is not enough. The top-most stationary lava block will reveal 2 top texels (not fill the block fully).

There are 2 ores that are visible this way using the default vanilla texture pack: Emerald and Nether Quartz ore. Not to mention custom modpacks that may make other ores visible in this case by drawing ore textures closer to the block edges.

Whether or not custom legitimate texture packs should be considered is up to debate. But I believe current patch will wrongly hide Emerald/Quartz ore, if so - it's hiding otherwise visible ores.

Reply:
The top 2 pixels was brought up on Discord, and ultimately it was disregarded as changing how anti x-ray works at all is not worth it for a not so common issue. You're always able to disable this feature and rather just use moderators.

Reply:
This is known and was already discussed. (See #2940.) Won't fix.